File: 275-h-index-II/hindex.ts
function hIndex(citations: number[]): number {
    citations.sort((a,b)=>b-a)
    let h;
    for(let i=0;i<citations.length;i++){
        console.log(citations[i], i+1)
        if(citations[i]<i+1) return i
        h=i+1
    }
    return h
};

File: 275-h-index-II/hindex.py
class Solution:
    def hIndex(self, citations):
        N = len(citations)
        buckets = [0] * (N + 1)
        
        for elem in citations:
            buckets[min(elem, N)] += 1
        
        accum = list(accumulate(buckets[1:][::-1]))[::-1]  
        compar = [accum[i] >= i + 1 for i in range(N)]  
        return (compar + [0]).index(0)      